### PR TITLE
Handle spread props in no-children-prop rule

### PR DESCRIPTION
**Pull request by randycoulman**
Nodes of type `ExperimentalSpreadProperty` don’t have a `key` property, causing the `no-children-prop` to raise a “Cannot read property ‘name’ of undefined” error.

Code such as:

```
return React.createElement(
  elementType,
  { className: classNames, ...otherProps },
  children
)
```

will trigger the error.

### CODE REVIEW

File: tests/lib/rules/no-children-prop.js [hunk 1]
```diff
@@ -15,6 +15,7 @@ var RuleTester = require('eslint').RuleTester;
 var parserOptions = {
   ecmaVersion: 6,
   ecmaFeatures: {
+    experimentalObjectRestSpread: true,
```
Comment (ljharb) [inline tests/lib/rules/no-children-prop.js]: let's not necessarily enable this for every test - can we enable it just for the new tests?

Comment (randycoulman) [inline tests/lib/rules/no-children-prop.js]: I thought about doing that, but in looking through all of the tests in the project, there isn't a single other case where different `parserOptions` are used for different test cases.  There are eight other test cases that add `experimentalObjectRestSpread: true` to their top-level `parserOptions`, though.

That said, if you think it's important to establish a new pattern here, I'm happy to make the change.

Comment (ljharb) [inline tests/lib/rules/no-children-prop.js]: I'm not 100% on it either way - it just seems like it'd be useful to test the parser with common (non-experimental) options.

Comment (randycoulman) [inline tests/lib/rules/no-children-prop.js]: I think you've got a valid point.

But I'm pretty big on code consistency, so I wonder if it would be better to do a separate PR with that change made in all of the cases like this.

Comment (ljharb) [inline tests/lib/rules/no-children-prop.js]: I'm not sure why - this is the PR that adds `experimentalObjectRestSpread`. but i don't think this needs to block if other maintainers think it's fine.

Comment (randycoulman) [inline tests/lib/rules/no-children-prop.js]: Yes, this PR adds it here.  But there are eight other tests that include `experimentalObjectRestSpread` in their default `parserOptions` just like this one does.

If it's important to only set that option for certain test cases, then I think we should make that consistent across all of the tests that use that option.  That's why I suggested a separate PR to make that change.

Comment (ljharb) [inline tests/lib/rules/no-children-prop.js]: Gotcha, that's a fair point.

**Comment by yannickcr** [comment]
Merged, thanks!

**Comment by ljharb** [comment]
@yannickcr any update on a release?

**Comment by yannickcr** [comment]
@ljharb sorry, I was pretty busy last days but I'll try to craft a release this week-end.
